R/hazard_link.R

```diff
@@ -0,0 +1,79 @@
+
+
+#' Hazard Link Class
+#'
+#' Creates a Hazard Link object which combines a longditudinal model to
+#' an OS model
+#'
+#' @slot stan A `StanModule` object as created by [StanModule()] that specifies any Stan code 
+#' required for the link / contribution
+#' @slot contribution Stan code for what this link contriutes to the log hazard in the OS model
+#' @slot parameter The name of link parameter
+#' @examples
+#' HazardLink(
+#'     parameter = "beta_ttg",
+#'     contribution = "beta_ttg * ttg(phi)",
+#'     stan = StanModule(
+#'         functions = "real ttg(real phi) {phi^2 };"
+#'     )
+#' )
+#' @export
+HazardLink <- setClass(
+    "HazardLink",
+    representation = list(
+        "stan" = "StanModule",
+        "contribution" = "character",
+        "parameters" = "character"
+    )
+)
+
+
+#' @importFrom assertthat assert_that
+#' @rdname StanModule-class
+#' @export
+setMethod(
+    f = "initialize",
+    signature = "HazardLink",
+    definition = function(.Object, ..., stan = StanModule(), contribution, parameters) {
+
+        assert_that(
+            is.character(parameters),
+            length(parameters) >= 1,
+            msg = "`parameter` must be a character vector"
+        )
+        
+        assert_that(
+            is.character(contribution),
+            length(contribution) == 1,
+            msg = "`contribution` must be length 1 character vectors"
+        )
+
```

Comment:
Suggestion: 
`assert_that(names(stan@priors) == parameters)`
Moreover, we can create a function to define a `normal(0,1)` prior for the link parameter if the user does not define the prior.

Reply:
I think we discussed this in yesterdays call but my feeling is that this might be too restrictive. For example it may be that users don't want to define priors and let stans default prior be used or perhaps they want to define priors for secondary parameters required for functions which aren't used in the hazard link (though this is likely bad practice).